source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,157 @@
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+
+      std::vector<Tensor<1,dim>> velocity_values(n_q_points);
+
+      // create a quadrature formula for the velocity for the surface of cells
+      const QGauss<dim-1> quadrature_formula_face (this->introspection().polynomial_degree.velocities+1);
+
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula_face,
+                                        update_values |
+                                        update_JxW_values |
+                                        update_quadrature_points);
+
+      std::vector<Tensor<1,dim>> surface_velocity_values (fe_face_values.n_quadrature_points);
+
+      double local_velocity_square_integral = 0;
+      double local_velocity_square_integral_top_boundary;
+      double local_top_boundary_area;
+      const std::set<types::boundary_id>
+      boundary_indicators
+        = this->get_geometry_model().get_used_boundary_indicators ();
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+            // extract velocities
+            fe_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                                                                                        velocity_values);
+            for (unsigned int q = 0; q < n_q_points; ++q)
+              {
+                local_velocity_square_integral += ((velocity_values[q] * velocity_values[q]) *
+                                                   fe_values.JxW(q));
+              }
+            //now for the surface cells only
+            for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+              if (cell->face(f)->boundary_id() == top_boundary_id)
+                {
+                  fe_face_values.reinit (cell, f);
+                  // extract velocities
+                  fe_face_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
+                      surface_velocity_values);
+
+                  // determine the squared velocity on the face
+                  double local_sqvel = 0.0;
+                  double local_fe_face_area = 0.0;
+                  for (unsigned int q = 0; q < n_q_points; ++q)
```

Comment:
This is what is causing the crash for the test. The volume integration has more integration points than the one over the surface, and calling fe_face_values.JxW(i) with an i larger than the maximum number of points crashes. That is why the test file was empty, but the error was posted here https://github.com/geodynamics/aspect/runs/4274792601?check_suite_focus=true (under `report test results`).
```suggestion
                  for (unsigned int q = 0; q < n_q_points_face; ++q)
```

Reply:
Ah yes! Thanks for the tip on finding the error